### DISCUSSION TITLE
gentoo 上 fcitx5 如何装 anthy?

### DISCUSSION THREAD

**Discussion by guyuming76** [fcitx/fcitx5#660] [Q&A]
接这个问题：https://github.com/fcitx/fcitx5/discussions/609

我gentoo linux 上已经装fcitx5了，如何才能装日文 anthy?

系统会提示我装 app-i18n/fcitx-4.2.9.8::gentoo-zh ，然后安装时告诉我文件有冲突。

**Comment by karuboniru** [comment]
Gentoo 源里面的 fcitx-anthy 应该是 fcitx4 的对应版本，你可能需要自己写 ebuilds 编译安装

**Reply by guyuming76** [reply]
我找到了这个： https://github.com/fcitx/fcitx5-anthy 
但clone下来，照着INSTALL文件，做到  ~/fcitx5-anthy/src/build $ cmake ..  这一步时，报错：

CMake Error at CMakeLists.txt:23 (fcitx5_translate_desktop_file):
  Unknown CMake command "fcitx5_translate_desktop_file".

**Reply by wengxt** [reply]
@guyuming76  请贴完整的错误，从上面的错误看你执行 cmake 命令的路径错了。cmake 的参数要是源码的根目录，而不是 src/ 目录

**Reply by wengxt** [reply]
https://fcitx-im.org/wiki/Compiling_fcitx5

**Reply by guyuming76** [reply]
我是照 https://github.com/fcitx/fcitx5-anthy/blob/master/INSTALL 做的

```
gym@gymDeskGentoo ~/fcitx5-anthy/src/build $ cmake ..
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at CMakeLists.txt:23 (fcitx5_translate_desktop_file):
  Unknown CMake command "fcitx5_translate_desktop_file".


CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.24)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
See also "/home/gym/fcitx5-anthy/src/build/CMakeFiles/CMakeOutput.log".
gym@gymDeskGentoo ~/fcitx5-anthy/src/build $
```

**Reply by wengxt** [reply]
……就是我说的问题，source directory 就是指解压后的根目录，不是 src/ 目录……

**Reply by guyuming76** [reply]
谢谢，这下 sudo make install 也成功了，fcitx5-configtool里选了个anthy后，ctrl+space 切换输入法，中文是可以输的，anthy弹出选词框后面有NotAvailable, 输入也没效果，是我少配了啥？
![20221123_10h09m46s_grim](https://user-images.githubusercontent.com/12045548/203457568-176bdfdc-cc6d-4695-b26e-a980ab6479db.png)

**Reply by wengxt** [reply]
仔细读文档，安装路径和系统的fcitx的路径不一致。

**Reply by wengxt** [reply]
关键词prefix

**Reply by guyuming76** [reply]
多谢，下面是我机器上fcitx5的安装情况：

```
gym@gymDeskGentoo ~ $ which fcitx5
/usr/bin/fcitx5
gym@gymDeskGentoo ~ $ equery belongs $(which fcitx5)
 * Searching for /usr/bin/fcitx5 ...
app-i18n/fcitx-5.0.19-r1 (/usr/bin/fcitx5)
```

我先是在~/fcitx5-anthy/build 目录 sudo make uninstall ，然后删掉build目录重建;

我若是 prefix=/usr/bin, 会发现fcitx5-configtools进去后连 anthy 都找不到。

观察下make install 安装的文件类型和机器上现有的文件，我设 prefix=/usr ，这次fcitx5-configtool就可以配置 anthy了。

但我发现在 DWM（xorg）和DWL(wayland)下，输入表现好像不太一样，我再摸索下

UPDATE: 我所说的不太一样 https://github.com/fcitx/fcitx5-anthy/issues/6
